pysatMadrigal/instruments/gnss_tec.py

```diff
@@ -149,27 +150,40 @@ def list_files(tag=None, inst_id=None, data_path=None, format_str=None,
         If filenames only store two digits for the year, then
         '1900' will be added for years >= two_digit_year_break
         and '2000' will be added for years < two_digit_year_break.
-    delimiter : string
+    delimiter : str
         Delimiter string upon which files will be split (e.g., '.')
-    file_type : string
-        File format for Madrigal data.  Load routines currently only accepts
-        'hdf5' and 'netCDF4', but any of the Madrigal options may be used
-        here. (default='netCDF4')
+    file_type : str or NoneType
+        File format for Madrigal data.  Load routines currently accepts 'hdf5',
+        'simple', and 'netCDF4', but any of the Madrigal options may be used
+        here. If None, will look for all known file types. (default=None)
 
     Returns
     -------
     out : pysat.Files.from_os : pysat._files.Files
         A class containing the verified available files
 
     """
-    if supported_tags[inst_id][tag].find('{file_type}') > 0:
-        supported_tags[inst_id][tag] = supported_tags[inst_id][tag].format(
-            file_type=file_type)
-
-    out = ps_gen.list_files(tag=tag, inst_id=inst_id, data_path=data_path,
-                            format_str=format_str, delimiter=delimiter,
-                            supported_tags=supported_tags,
-                            two_digit_year_break=two_digit_year_break)
+    file_types = general.file_types.keys() if file_type is None else [file_type]
```

Comment:
Since this function is repeated should we make it a general function?

Reply:
Huh, I remember looking at this and thinking it wouldn't be practical, but it was super easy.